src/client/Microsoft.Identity.Client/Platforms/Android/AndroidBroker.cs

```diff
@@ -160,17 +160,32 @@ internal static void SetBrokerResult(Intent data, int resultCode)
                     return;
                 }
 
-                if (resultCode != (int)BrokerResponseCode.ResponseReceived)
+                switch (resultCode)
                 {
-                    s_androidBrokerTokenResponse = new MsalTokenResponse
-                    {
-                        Error = MsalError.BrokerResponseReturnedError,
-                        ErrorDescription = data.GetStringExtra(BrokerConstants.BrokerResultV2),
-                    };
-                }
-                else
-                {
-                    s_androidBrokerTokenResponse = CreateMsalTokenResponseFromResult(data.GetStringExtra(BrokerConstants.BrokerResultV2));
+                    case (int)BrokerResponseCode.ResponseReceived:
+                        s_androidBrokerTokenResponse = CreateMsalTokenResponseFromResult(data.GetStringExtra(BrokerConstants.BrokerResultV2));
+                        break;
+                    case (int)BrokerResponseCode.UserCancelled:
+                        s_androidBrokerTokenResponse = new MsalTokenResponse
+                        {
+                            Error = MsalError.AuthenticationCanceledError,
```

Comment:
I don't see any changes to the iOS broker - assume this is separate PR. I remember that iOS broker does not handle this case explicitly, it just throws whatever error code it gets from the broker. 

So there are 2 ways of fixing this: 

1. make changes to the iOS broker so that it throws this exact error code 
2. make changes to the Android broker so that it throws the same error code with ios

Personally I like option 1 more because it's closer to webview error code. However it's not exactly backwards compatible. Option 2 is more backwards compatible.